src/iop/zonesystem.c

```diff
@@ -203,7 +203,7 @@ static void process_common_cleanup(struct dt_iop_module_t *self, dt_dev_pixelpip
   const int ch = piece->colors;
   const int size = d->params.size;
 
-  if(piece->pipe->mask_display) dt_iop_alpha_copy(ivoid, ovoid, width, height);
+  if(piece->pipe->mask_display & DT_DEV_PIXELPIPE_DISPLAY_MASK) dt_iop_alpha_copy(ivoid, ovoid, width, height);
```

Comment:
> so it seems to be always done in the kernel

Yes

> But dt_iop_alpha_copy() is pretty much always the case for cpu codepath...

This should ideally be handled without the need of dt_iop_alpha_copy() just by the processing code like in the OpenCL kernels. But several modules don't do it like that. They need dt_iop_alpha_copy(). That's not a big deal. Let's leave it at that, especially to keep some identity to what a process_cl() does versus process()/process_sse2().